metpy/interpolate/grid.py

```diff
@@ -9,6 +9,7 @@
 
 from .points import (interpolate_to_points, inverse_distance_to_points,
                      natural_neighbor_to_points)
+from .. import calc as mpcalc
```

Comment:
So Python 2.7 hates this because it leads to a circular import. `calc.tools` imports some interpolation functions from `interpolate.one_dimension`, which triggers an import of `interpolate` -> `interpolate.grid`, which goes back to `calc`. Even worse it goes back to `calc.tools`.

The easiest way I can think to break the loop for 2.7 is to put this import in the function below:
```python
from ..calc import find_bounding_indices
```

Hell, even leave it as a note to fix once we drop 2.7.

Reply:
I run into a problem trying to import in this way using Python 3 because of a similar loop and I get the following error ...

```python---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-313f2d808e2b> in <module>()
      4 import cartopy.feature as cfeature
      5 import matplotlib.pyplot as plt
----> 6 import metpy.calc as mpcalc
      7 from metpy.units import units
      8 import numpy as np

~/unidata_stuff/MetPy/metpy/calc/__init__.py in <module>()
      5 
      6 from .basic import *  # noqa: F403
----> 7 from .cross_sections import *  # noqa: F403
      8 from .indices import *  # noqa: F403
      9 from .kinematics import *  # noqa: F403

~/unidata_stuff/MetPy/metpy/calc/cross_sections.py in <module>()
     12 
     13 from .basic import coriolis_parameter
---> 14 from .tools import first_derivative
     15 from ..package_tools import Exporter
     16 from ..xarray import CFConventionHandler, check_matching_coordinates

~/unidata_stuff/MetPy/metpy/calc/tools.py in <module>()
     23 from ..cbook import broadcast_indices
     24 from ..deprecation import deprecated, metpyDeprecation
---> 25 from ..interpolate.one_dimension import interpolate_1d, interpolate_nans_1d, log_interpolate_1d
     26 from ..package_tools import Exporter
     27 from ..units import atleast_1d, check_units, concatenate, diff, units

~/unidata_stuff/MetPy/metpy/interpolate/__init__.py in <module>()
      4 r"""Provides tools for interpolating data."""
      5 
----> 6 from .grid import *  # noqa: F403
      7 from .one_dimension import *  # noqa: F403
      8 from .points import *  # noqa: F403

~/unidata_stuff/MetPy/metpy/interpolate/grid.py in <module>()
     10 from .points import (interpolate_to_points, inverse_distance_to_points,
     11                      natural_neighbor_to_points)
---> 12 from ..calc import find_bounding_indices
     13 from ..deprecation import deprecated
     14 from ..package_tools import Exporter

ImportError: cannot import name 'find_bounding_indices'```

Reply:
The following 
```python
from .. import calc
```
worked and then the function was called with
```python
calc.find_bounding_indices()
```

should work for the time being provided that it is liked by Python 2.7

Reply:
Nope, doesn't work for Python 2.7. I just created an environment with 2.7 to try to work it out.

Reply:
Okay, I think I got it to work now. ```import metpy.calc```